__tests__/endpoints.test.js

```diff
@@ -96,3 +95,53 @@ describe("/api", () => {
       });
   });
 });
+
+describe("/api/articles/:article_id", () => {
+  test("test for 200 status on completion", () => {
+    return request(app).get(`/api/articles/3`).expect(200);
+  });
+  test("should have the correct article by id, with correct amount of properties", () => {
+    return request(app)
+      .get(`/api/articles/1`)
+      .expect(200)
+      .then((result) => {
+        expect(Object.keys(result.body).length).toBe(8);
+      });
+  });
+  test("should have all 8 properties", () => {
+    return request(app)
+      .get(`/api/articles/1`)
+      .expect(200)
+      .then((result) => {
+        expect(result.body).toHaveProperty("article_id");
+        expect(result.body).toHaveProperty("title");
+        expect(result.body).toHaveProperty("topic");
+        expect(result.body).toHaveProperty("author");
+        expect(result.body).toHaveProperty("body");
+        expect(result.body).toHaveProperty("created_at");
+        expect(result.body).toHaveProperty("votes");
+        expect(result.body).toHaveProperty("article_img_url");
+      });
+  });
+  test("all properties should be of the correct data type", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then((result) => {
+        expect(typeof result.body.article_id).toBe("number");
+        expect(typeof result.body.title).toBe("string");
+        expect(typeof result.body.topic).toBe("string");
+        expect(typeof result.body.author).toBe("string");
+        expect(typeof result.body.body).toBe("string");
+        expect(typeof result.body.created_at).toBe("string");
+        expect(typeof result.body.votes).toBe("number");
+        expect(typeof result.body.article_img_url).toBe("string");
+      });
+  });
+  test('should return 404 not found for incorrect path', () => {
+    return request(app).get('/api/articles/33').expect(404);
```

Comment:
Can we check the response body for a relevant message for these two test please :)

Good endpoint tests and error codes though!